### DISCUSSION TITLE
refactor: 스타카토 저장 버튼 위치 수정

### DISCUSSION THREAD

**Discussion by BurningFalls** [woowacourse-teams/2024-staccato#713] [Feature]
### 스타카토 생성 화면

<img src="https://github.com/user-attachments/assets/f5c7e553-7c32-4457-a443-b2a160b3dd5a" width="30%" height="30%">
<img src="https://github.com/user-attachments/assets/fb0ab378-9513-42f7-88df-0a72876051e2" width="30%" height="30%">

### 스타카토 화면

<img src="https://github.com/user-attachments/assets/1324c2a0-2321-4a47-a031-29fa66c35d8a" width="30%" height="30%">

### 스타카토 수정 화면

<img src="https://github.com/user-attachments/assets/bbbc247a-b87a-4df1-a157-34fe6e45e9e6" width="30%" height="30%">
<img src="https://github.com/user-attachments/assets/2e7fd525-05c8-416c-a9d1-2254c6a9640f" width="30%" height="30%">

<br><br>

* 스타카토 수정에서 주로 수정하는 내용은 사진인데, 사진 수정 후 `저장` 버튼을 누르려고 아래로 드래그해야 하는 점이 불편하다는 사용자 피드백이 있었습니다.
* `저장` 버튼을 오른쪽 상단에 **고정**시키면(스크롤을 움직여도 저장 버튼은 고정되어 있음), 위와 같은 불편함을 해소할 수 있다고 생각합니다.

### 추가 의견

< 의견 1 >

스타카토 생성의 경우, 아래 내용을 입력하려고 어차피 내리기 때문에 굳이 저장 버튼이 위에 있을 필요는 없다?
* 사용자가 다른 내용을 먼저 입력하고 나중에 사진을 등록한다고 생각하면, 저장 버튼이 위에 고정되어 있는 게 좋을 것 같습니다.
* 수정 화면과 통일성을 가져가기 위해서도 저장 버튼을 상단에 고정해도 좋을 것 같습니다.

## 🤩 To Be
> 리팩터링 방향을 구체적으로 공유해주세요.

논의 후 결정

## 😇 이때까지 끝낼게요!
> 리팩터링 완료 예상 날짜를 작성해주세요. 신중하게 생각해요!

논의 후 결정

## 🙇‍♀️이슈 확인했어요:)
> 팀원에게 이슈 확인을 부탁해요!
- [x] @Junyoung-WON 
- [x] @hxeyexn 
- [x] @s6m1n 
- [x] @linirini 
- [ ] @Ho-Tea

**Comment by linirini** [comment]
상단바?는 어차피 스크롤해도 계속 있기 때문에 스타카토 생성/수정 화면 분리할 필요 없이 상단 바 우측에 저장 버튼을 배치하는 방식으로 통일해도 괜찮을 것 같습니다!

**Comment by Junyoung-WON** [comment]
저도 상단 바 우측에 저장 버튼을 배치하는 것이 좋을 것 같습니다!
눈에 잘 띄고, 스크롤을 안해도 된다는 장점이 크네요!

다만 지금 구현된 저장 버튼처럼, 사용자에게 저장 가능 여부를 버튼 UI의 모습을 바꾸어서 보여주어야 합니다.
저장 버튼을 우측 상단으로 배치할 경우, 저장 버튼의 위치, 크기가 달라지기 때문에 디자인 수정이 필요할 수 있습니다.
위 의견이 받아들여진다면, 어떤 디자인이 좋을지는 안드로이드 측에서 고민해보고 시안을 만들어보아야 할 것 같아요~

**Comment by hxeyexn** [comment]
> 다만 지금 구현된 저장 버튼처럼, 사용자에게 저장 가능 여부를 버튼 UI의 모습을 바꾸어서 보여주어야 합니다.

툴바(상단바) 우측에 저장 버튼을 배치하는 것도 좋지만 버튼 활성, 비활성 여부를 툴바(상단바)에서 나타내는 게 쉽지 않을 것 같다는 생각이 드네요..! **저장 버튼을 고정시키거나 하단으로 이동하는 플로팅 버튼을 추가하는 방법은 어떨까요?**

### 방법 1. 저장 버튼 고정 
- 화면 스크롤과 관계없이 버튼이 보이도록 고정

|화면 스크롤 하지 않았을 때| 화면 스크롤 했을 때|
|:---:|:---:|
|<img src="https://github.com/user-attachments/assets/569c70ad-0383-4955-a3d2-c8d95aa746cf" width="260">|<img src="https://github.com/user-attachments/assets/8204527d-b29f-4a32-b4bc-2902de8fdf5a" width="260">|

### 방법 2. 화면 하단으로 이동하는 플로팅 버튼 추가
- 화면 상단에 있을 때 화면 맨 아래로 이동하는 플로팅 버튼을 띄워줌
- 플로팅 버튼 클릭 시 화면 맨 아래로 이동

|플로팅 버튼 클릭 전|플로팅 버튼 클릭 후(화면 맨 아래로 이동)|
|:---:|:---:|
|<img src="https://github.com/user-attachments/assets/00156706-6baf-4777-81c5-d343472cc7c3" width="260">|<img src="https://github.com/user-attachments/assets/1a2d0fe3-bd76-448c-88cf-a044a71f1a41" width="260">|

**Comment by linirini** [comment]
> 툴바(상단바) 우측에 저장 버튼을 배치하는 것도 좋지만 버튼 활성, 비활성 여부를 툴바(상단바)에서 나타내는 게 쉽지 않을 것 같다는 생각이 드네요..! **저장 버튼을 고정시키거나 하단으로 이동하는 플로팅 버튼을 추가하는 방법은 어떨까요?**
> 
> ### 방법 1. 저장 버튼 고정
> * 화면 스크롤과 관계없이 버튼이 보이도록 고정
>
> ### 방법 2. 화면 하단으로 이동하는 플로팅 버튼 추가
> * 화면 상단에 있을 때 화면 맨 아래로 이동하는 플로팅 버튼을 띄워줌
> * 플로팅 버튼 클릭 시 화면 맨 아래로 이동

둘 다 좋은데요????👍👍

**Comment by BurningFalls** [comment]
와 방법1 너무 좋아요! 저장 버튼을 오른쪽 위에 두는 것으로만 고정관념에 사로잡혀 있었는데, 저렇게 하면 기존 디자인도 살리는 방향으로 가져갈 수 있어서, 원래 저희 앱을 사용하던 사용자들에게도 불편함 없는 업데이트를 제공할 수 있을 것 같아요!
